src/plugins/apimon/apimon.cpp

```diff
@@ -162,61 +138,31 @@ static event_response_t usermode_return_hook_cb(drakvuf_t drakvuf, drakvuf_trap_
-            break;
-        case OUTPUT_KV:
-            printf("apimon Time=" FORMAT_TIMEVAL ",VCPU=%" PRIu32 ",CR3=0x%" PRIx64 ",PID=%d,PPID=%d,ProcessName=\"%s\",UserID=%" PRIi64 ",Method=\"%s\",CalledFrom=0x%" PRIx64 ",ReturnValue=0x%" PRIx64 ",Arguments=",
-            UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3,info->proc_data.pid, info->proc_data.ppid, info->proc_data.name,
-            info->proc_data.userid, info->trap->name, info->regs->rip, info->regs->rax);
-            print_arguments(drakvuf, info, ret_target->arguments, ret_target->argument_printers);
-            break;
-        case OUTPUT_JSON:
-            escaped_pname = drakvuf_escape_str(info->proc_data.name);
-            printf( "{"
-                    "\"Plugin\": \"apimon\", "
-                    "\"Event\": \"api_called\", "
-                    "\"TimeStamp\":" "\"" FORMAT_TIMEVAL "\", "
-                    "\"ProcessName\": %s, "
-                    "\"UserName\": \"%s\", "
-                    "\"UserId\": %" PRIu64 ", "
-                    "\"PID\": %d, "
-                    "\"PPID\": %d, "
-                    "\"TID\": %d, "
-                    "\"Method\": \"%s\", "
-                    "\"CalledFrom\": \"0x%" PRIx64 "\", "
-                    "\"ReturnValue\": \"0x%" PRIx64 "\", "
-                    "\"Arguments\": ",
-                    UNPACK_TIMEVAL(info->timestamp),
-                    escaped_pname,
-                    USERIDSTR(drakvuf), info->proc_data.userid,
-                    info->proc_data.pid, info->proc_data.ppid, info->proc_data.tid,
-                    info->trap->name,
-                    info->regs->rip,
-                    info->regs->rax);
-
-            print_arguments(drakvuf, info, ret_target->arguments, ret_target->argument_printers);
-            printf(", "
-                   "\"Extra\": {");
-            print_extra_data(extra_data);
-            printf("}}");
-            g_free(escaped_pname);
-            break;
-        default:
-        case OUTPUT_DEFAULT:
-            printf("[APIMON-USERHOOK] TIME:" FORMAT_TIMEVAL " VCPU:%" PRIu32 " CR3:0x%" PRIx64 " ProcessName:\"%s\" UserID:%" PRIi64 " Method:\"%s\" CalledFrom:0x%" PRIx64 " ReturnValue:0x%" PRIx64 " Arguments:",
-            UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
-            info->proc_data.userid, info->trap->name, info->regs->rax, info->regs->rip);
-            print_arguments(drakvuf, info, ret_target->arguments, ret_target->argument_printers);
-            break;
+        const auto &args = ret_target->arguments;
+        const auto &printers = ret_target->argument_printers;
+        auto arg = std::cbegin(args);
+        auto printer = std::cbegin(printers);
+        for (; arg != std::cend(args) && printer != std::cend(printers); ++arg, ++printer) {
+            fmt_args.push_back(fmt::Qstr((*printer)->print(drakvuf, info, *arg)));
+        }
```

Comment:
```suggestion
        const auto &args = ret_target->arguments;
        const auto &printers = ret_target->argument_printers;
        for (auto [arg, printer] = std::tuple(std::cbegin(args), std::cbegin(printers)); arg != std::cend(args) && printer != std::cend(printers); ++arg, ++printer) {
            fmt_args.push_back(fmt::Qstr((*printer)->print(drakvuf, info, *arg)));
        }
```

Reply:
Yes, that was my first implementation, however long lines == bad.
This is the cause of putting it all inside a block.

Reply:
It's still 25% shorter than previous printing code.

Reply:
```c++
        for (auto [arg, printer] = std::tuple(std::cbegin(args), std::cbegin(printers));
             arg != std::cend(args) && printer != std::cend(printers);
             ++arg, ++printer) {
```
we can format it like this :thinking:

Reply:
Imo it's better this way, because if not, people will see a variable outside of for loop and start wondering why it wasn't put in for loop. Conclusion *because line would have been too long* isn't my first idea.